packages/esm-patient-vitals-app/src/vitals-biometrics-form/delete-vitals-biometrics-modal.component.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Button, ModalBody, ModalHeader, ModalFooter } from '@carbon/react';
+import { showSnackbar } from '@openmrs/esm-framework';
+import {
+  deleteVitalsAndBiometrics,
+  getEncounterByUuid,
+  invalidateCachedVitalsAndBiometrics,
+} from '../common/data.resource';
+
+interface DeleteVitalModalProps {
+  closeDeleteModal: () => void;
+  encounterUuid: string;
+  patientUuid: string;
+}
+
+const DeleteVitalsBiometricsModal = ({ closeDeleteModal, encounterUuid }: DeleteVitalModalProps) => {
+  const { t } = useTranslation();
+  const [isDeleting, setIsDeleting] = useState(false);
+
+  const voidVitalsInEncounter = async (encounterUuid: string) => {
+    try {
+      const response = await getEncounterByUuid(encounterUuid);
+      if (!response?.data) {
+        showSnackbar({
+          isLowContrast: false,
+          kind: 'error',
+          title: t('encounterNotFound', 'Encounter not found'),
+        });
+      }
+
+      const updatedEncounter = {
+        ...response.data,
+        obs: response.data.obs?.map((obs) => ({ ...obs, voided: true })) || [],
```

Comment:
This is still incorrect, in that it assumes all `obs` in the encounter are vitals, which cannot always be guaranteed to be the case.

Reply:
What should happen here instead, @ibacher?

Reply:
I think we have to look at the obs concept and make sure it's one in the set of vitals.

Reply:
See [this](https://github.com/openmrs/openmrs-esm-patient-chart/pull/1812#discussion_r1639704029) below.